### PR TITLE
improve range slider filter

### PR DESCRIPTION
**Pull request by zlatanpham**
This PR aims to make improvements in performance and usability of the `rangeAggregrateFilter` and its UI wrapper `RangeAggregrateSlider`:

- put `filter.set` inside a denounce to limit the number of search call
- hide the slider if the `min` and `max` are equal
- auto join `field` value into `min` and `max` property of `Value` object
- remove unused `multi` field from `rangeAggregrateFilter` constructor

### CODE REVIEW

File: src/controllers/rangeAggregateFilter.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+    const min = (aggregates[`min.${this._field}`] as number) || 0;
+    const max = (aggregates[`max.${this._field}`] as number) || 0;
+
+    return [min, max];
+  }
+
+  private _addMinMaxToValues(values: Values) {
+    const min = values.get().min;
+    const max = values.get().max;
```
[the remark this answers is not in the document]
Comment (zlatanpham) [reply src/controllers/rangeAggregateFilter.ts:90]: > Can we not use destructuring here? Also prevents a call to `.get()`.

Hey @sampotts, you meant it should be like that:
```jsx
const {min, max} = values.get();
```

[the remark this answers is not in the document]
Comment (sampotts) [reply src/controllers/rangeAggregateFilter.ts:90]: Yeah. Just a suggestion really. Not major.

[the remark this answers is not in the document]
Comment (zlatanpham) [reply src/controllers/rangeAggregateFilter.ts:90]: > As an aside, do we have linting and prettier configs for stuff like this?

You mean by preventing from destructuring from a method call like `min = values.get().min`? If true, I don't think `eslint` had a rule like that.

[the remark this answers is not in the document]
Comment (sampotts) [reply src/controllers/rangeAggregateFilter.ts:90]: image

I often use the Airbnb config. I noticed there's a few TypeScript versions of it. It's something I'll take a look at soon. Don't worry too much about it for now.

**Comment by zlatanpham** [comment]
Yes. we need @benhinchley to take notice and publish the new version as well. Or could you do that @sampotts?

**Comment by sampotts** [comment]
> Yes. we need @benhinchley to take notice and publish the new version as well. Or could you do that @sampotts?

I have access now. I'll check with Ben what the process is there 👍

**Comment by sampotts** [comment]
@zlatanpham Apologies for the delay. This has been deployed in v2.8.15 👍

**Pull request merged**